src/pages/_app.tsx

```diff
@@ -33,6 +34,23 @@ const { ConsentManager, openConsentManager } = createConsentManager({
 export default function App({ Component, pageProps, layoutProps }) {
   useAnchorLinkAnalytics()
 
+  /**
+   * TODO: this will be temporary and last for as long as we have to manually
+   * hydrate product data from page components. Ideally, we won't have to do
+   * this we use more dynamic routes.
+   */
+  const currentPath = useCurrentPath({ excludeHash: true, excludeSearch: true })
+  const isDevDotPage =
+    !currentPath.startsWith(`/_proxied-dot-io`) &&
+    !currentPath.startsWith(`/swingset`)
+  const isNotHomePage = currentPath !== '/'
+  const isMissingProductData = !pageProps.product
+  if (isDevDotPage && isNotHomePage && isMissingProductData) {
+    throw new Error(
+      `\`product\` property is missing from \`pageProps\` prop on path: \`${currentPath}\`. Product data is needed to hydrate \`CurrentProductContext\`.`
+    )
```

Comment:
Suggestion: Where can we get the data if it's missing? Maybe mention that the data can be read from `src/data/{product}.json`

Reply:
Great idea! I'll put that in the message.

Reply:
Updated in: https://github.com/hashicorp/dev-portal/pull/222/commits/346090bf5a9349344b7bb6428d79231d31cd51ba